src/include/migraphx/op/dequantizelinear.hpp

```diff
@@ -49,10 +49,13 @@ struct dequantizelinear
     std::string name() const { return "dequantizelinear"; }
     shape compute_shape(std::vector<shape> inputs) const
     {
-        check_shapes{inputs, *this}.same_dims().has(2, 3);
-        if(inputs.size() == 3 and inputs[0].type() != inputs[2].type())
+        if(inputs.size() == 3)
```

Comment:
Thanks for fixing this, but perhaps this Operator is still missing an important check when there are 2 or 3  inputs. And `x_scale` is a 1-D Tensor. We should check this 1-D Tensor against the axis dimension of `X`. I don't see that check in parse_dequantizelinear.cpp either. Is it done somewhere else? Thanks.

Reply:
I believe it should be. If not I'll make another issue and PR for that case

Reply:
> And x_scale is a 1-D Tensor. We should check this 1-D Tensor against the axis dimension of X.

No, this is not the onnx `DequantizeLinear` operator. In migraphx, all inputs should have the same dimensions(which is what `check_shapes` checks for). If in onnx, the `x_scale` is a 1-D tensor then a broadcast is inserted to make it have the same dimensions here:

https://github.com/ROCm/AMDMIGraphX/blob/fab88b0eca40b85d8bd53a40ff75acc587ed019d/src/onnx/quantize_dequantize_linear.cpp#L70

Reply:
Thanks, Paul. For the onnx case, I do see the 1D tensor shape is being checked here: [quantize_dequantize_linear.cpp](https://github.com/ROCm/AMDMIGraphX/blob/fab88b0eca40b85d8bd53a40ff75acc587ed019d/src/onnx/quantize_dequantize_linear.cpp#L60)